consensus/src/context/rx_vms.rs

```diff
@@ -125,64 +125,72 @@ impl RandomXVMCache {
-        let should_make_vms = hf >= &HardFork::V12;
-        if should_make_vms && self.vms.len() != self.seeds.len() {
-            // this will only happen when syncing and rx activates.
-            tracing::debug!("RandomX has activated, initialising VMs");
-
-            let seeds_clone = self.seeds.clone();
-            self.vms = rayon_spawn_async(move || {
-                seeds_clone
-                    .par_iter()
-                    .map(|(height, seed)| {
-                        (
-                            *height,
-                            Arc::new(RandomXVM::new(seed).expect("Failed to create RandomX VM!")),
-                        )
-                    })
-                    .collect()
-            })
-            .await
-        }
-
-        if is_randomx_seed_height(height) {
-            tracing::debug!("Block {height} is a randomX seed height, adding it to the cache.",);
-
-            self.seeds.push_front((height, *hash));
-
-            if should_make_vms {
                 let new_vm = 'new_vm_block: {
                     tracing::debug!(
-                        "Past hard-fork 12 initializing VM for seed: {}",
-                        hex::encode(hash)
+                        "Initializing RandomX VM for seed: {}",
+                        hex::encode(next_seed_hash)
                     );
 
                     // Check if we have been given the RX VM from another part of Cuprate.
                     if let Some((cached_hash, cached_vm)) = self.cached_vm.take() {
-                        if &cached_hash == hash {
+                        if cached_hash == next_seed_hash {
                             tracing::debug!("VM was already created.");
                             break 'new_vm_block cached_vm;
                         }
                     };
 
-                    let hash_clone = *hash;
-                    rayon_spawn_async(move || Arc::new(RandomXVM::new(&hash_clone).unwrap())).await
+                    rayon_spawn_async(move || Arc::new(RandomXVM::new(&next_seed_hash).unwrap()))
+                        .await
                 };
 
-                self.vms.insert(height, new_vm);
+                self.vms.insert(seed_height, new_vm);
             }
+            // More than one more seed than VM.
+            _ => {
+                // this will only happen when syncing and rx activates.
+                tracing::debug!("RandomX has activated, initialising VMs");
+
+                let seeds_clone = self.seeds.clone();
+                self.vms = rayon_spawn_async(move || {
+                    seeds_clone
+                        .par_iter()
+                        .map(|(height, seed)| {
+                            (
+                                *height,
+                                Arc::new(
+                                    RandomXVM::new(seed).expect("Failed to create RandomX VM!"),
+                                ),
+                            )
```

Comment:
nit: less indentation
```suggestion
                        .map(|(height, seed)| {
                            let vm = RandomXVM::new(seed).expect("Failed to create RandomX VM!");
                            let vm = Arc::new(vm);
                            (*height, vm)
                        })
```